src/frontend/js/components/Modal/index.tsx

```diff
@@ -69,26 +69,28 @@ export const Modal = ({
       overlayClassName={mergeClasses({ base: 'modal__overlay', classes: overlayClassName })}
       {...props}
     >
-      <header className={headerClasses.join(' ')}>
-        {StringHelper.isString(title) && <h2>{title}</h2>}
-        {!StringHelper.isString(title) && title}
-        {hasCloseButton && (
-          <Button
-            aria-label={intl.formatMessage(messages.closeDialog)}
-            className="modal__closeButton"
-            onClick={(e) => props.onRequestClose?.(e)}
-            title={intl.formatMessage(messages.closeDialog)}
-            color="tertiary"
-            size="small"
-          >
-            <Icon name={IconTypeEnum.ROUND_CLOSE} />
-            <span className="offscreen">
-              <FormattedMessage {...messages.closeDialog} />
-            </span>
-          </Button>
-        )}
-      </header>
-      {children}
+      <CunninghamProvider>
```

Comment:
I think the issue is more global. You should declare the CunninghamProvider at the top level of the React app https://github.com/openfun/richie/blob/56fa25c9119168f3329cf43d54eb96df3c279857/src/frontend/js/index.tsx#L115-L123